packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,603 @@
+    return new Connection(arn);
+  }
+  /**
+   * The ARN of the Connection for App Runner service to connect to the repository.
+   */
+  public readonly connectionArn: string
+  constructor(arn: string) {
+    this.connectionArn = arn;
+  }
+}
+
+/**
+ * Attributes for the App Runner Service
+ */
+export interface ServiceAttributes {
+  /**
+   * The name of the service.
+   */
+  readonly serviceName: string;
+  /**
+   * The ARN of the service.
+   */
+  readonly serviceArn: string;
+  /**
+   * The URL of the service.
+   */
+  readonly serviceUrl: string;
+  /**
+   * The status of the service.
+   */
+  readonly serviceStatus: string;
+}
+
+/**
+ * Represents the App Runner Service.
+ */
+export interface IService extends cdk.IResource {
+  /**
+   * The Name of the service.
+   */
+  readonly serviceName: string;
+  /**
+   * The ARN of the service.
+   */
+  readonly serviceArn: string;
+}
+
+/**
+ * The App Runner Service.
+ */
+export class Service extends cdk.Resource {
+  /**
+   * Import from service name.
+   */
+  public static fromServiceName(scope: Construct, id: string, serviceName: string): IService {
+    class Import extends cdk.Resource {
+      public serviceName = serviceName;
+      public serviceArn = cdk.Stack.of(this).formatArn({
+        resource: 'service',
+        service: 'apprunner',
+        resourceName: serviceName,
+      })
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * Import from service attributes.
+   */
+  public static fromServiceAttributes(scope: Construct, id: string, attrs: ServiceAttributes): IService {
+    const serviceArn = attrs.serviceArn;
+    const serviceName = attrs.serviceName;
+    const serviceUrl = attrs.serviceUrl;
+    const serviceStatus = attrs.serviceStatus;
+
+    class Import extends cdk.Resource {
+      public readonly serviceArn = serviceArn
+      public readonly serviceName = serviceName
+      public readonly serviceUrl = serviceUrl
+      public readonly serviceStatus = serviceStatus
+    }
+
+    return new Import(scope, id);
+  }
+  private readonly props: ServiceProps;
+  private accessRole?: iam.IRole;
+  private ecrRepository?: ecr.IRepository;
+  /**
+   * The ARN of the Service.
+   * @attribute
+   */
+  readonly serviceArn: string;
+  /**
+   * The ID of the Service.
+   * @attribute
+   */
+  readonly serviceId: string;
+  /**
+   * The URL of the Service.
+   * @attribute
+   */
+  readonly serviceUrl: string;
+  /**
+   * The status of the Service.
+   * @attribute
+   */
+  readonly serviceStatus: string;
+  /**
+   * The name of the service.
+   * @attribute
+   */
+  readonly serviceName: string;
+  constructor(scope: Construct, id: string, props: ServiceProps = {}) {
+    super(scope, id);
+    // either image or code is allowed. Not both.
+    if ((!props.code && !props.image) || (props.code && props.image)) {
+      throw new Error('Either image or code is required, not both.');
+    }
```

Comment:
Looks like this could be amenable to using an enum-like factory (`Code.fromImage` / `Code.fromGitHub` / ...) so this validation becomes redundant.

Reply:
Interesting.

Then we will have:

Code.fromEcrPublic()
Code.fromEcrRepository()
Code.fromGitHub()
Code.fromDockerImageAssets()

And then we can remove the `image` construct property. Right?

Reply:
Hi @RomainMuller,

How about experience like this?


```ts
// Scenario 1: Create the service from ECR public
const service1 = new Service(stack, 'Service1', {
  code: Code.fromImage(ContainerImage.fromEcrPublic('public.ecr.aws/aws-containers/hello-app-runner:latest')),
  port: 80,
});

// Scenario 2: Create the service from existing ECR repository, make sure you have `nginx` ECR repo in your account.
const repo = ecr.Repository.fromRepositoryName(stack, 'NginxRepository', 'nginx');
const service2 = new Service(stack, 'Service2', {
  code: Code.fromImage(ContainerImage.fromEcrRepository(repo)),
  port: 80,
});

// Scenario 3: Create the service from local code assets
const imageAssets = new assets.DockerImageAsset(stack, 'ImageAssets', {
  directory: path.join(__dirname, './docker.assets'),
});
const service3 = new Service(stack, 'Service3', {
  code: Code.fromImage(ContainerImage.fromDockerImageAssets(imageAssets)),
  port: 80,
});

// Scenario 4: Create the service from Github. Make sure you specify a valid connection ARN.
const connectionArn = stack.node.tryGetContext('CONNECTION_ARN') || 'MOCK';
const service4 = new Service(stack, 'Service4', {
  gitHubConnection: GitHubConnection.fromConnectionArn(connectionArn),
  code: Code.fromGitHub({
    repositoryUrl: 'https://github.com/aws-containers/hello-app-runner',
    branch: 'main',
    runtime: CodeRuntime.PYTHON_3,
  }),
});
```

Reply:
Hi @RomainMuller 

Any comments on this? Let's get it sorted on the high level design before I update this PR.

Thanks.